ros2interface/ros2interface/verb/show.py

```diff
@@ -14,12 +14,102 @@
 
 import argparse
 import sys
+import typing
 
 from ros2interface.api import type_completer
 from ros2interface.verb import VerbExtension
+from rosidl_adapter.parser import \
+    parse_message_string, MessageSpecification, \
+    SERVICE_REQUEST_RESPONSE_SEPARATOR, ACTION_REQUEST_RESPONSE_SEPARATOR, \
+    Field
 from rosidl_runtime_py import get_interface_path
 
 
+class InterfaceTextLine:
+    """
+    This class uses the class MessageSpecification to parse a single line
+    from an interface file and makes the type of the message accessible so
+    that it can be used to get the file path to nested messages.
+    """
+
+    def __init__(
+            self,
+            pkg_name: str,
+            msg_name: str,
+            line_text: str,
+    ):
+        if line_text in (SERVICE_REQUEST_RESPONSE_SEPARATOR, ACTION_REQUEST_RESPONSE_SEPARATOR):
+            msg_spec = None
+        else:
+            msg_spec = parse_message_string(
+                pkg_name=pkg_name,
+                msg_name=msg_name,
+                message_string=line_text,
+            )
+            if len(msg_spec.fields) > 1:
+                raise ValueError("'line_text' must be only one line")
+        self._msg_spec: typing.Optional[MessageSpecification] = msg_spec
+        self._raw_line_text = line_text
+
+    def __str__(self) -> str:
+        return self._raw_line_text
+
+    @property
+    def nested_type(self) -> typing.Optional[str]:
+        if self._field and self._is_nested():
+            interface_type: str = str(self._field.type)
+            if self._field.type.is_array:
+                interface_type = interface_type[:interface_type.find("[")]
+            return interface_type.replace("/", "/msg/")
+
+    @property
+    def _field(self) -> typing.Optional[Field]:
+        if self._msg_spec and self._msg_spec.fields:
+            return self._msg_spec.fields[0]
+
+    def _is_nested(self) -> bool:
+        if self._msg_spec and self._msg_spec.fields:
+            return "/" in str(self._field.type)
+        else:
+            return False
+
+
+def _get_interface_lines(interface_identifier: str) -> typing.Iterable[InterfaceTextLine]:
+
+    parts: typing.List[str] = interface_identifier.split("/")
+    if len(parts) != 3:
+        raise ValueError(f"Invalid name '{interface_identifier}'. Expected three parts separated by '/'")
+    pkg_name, _, msg_name = parts
+
+    file_path = get_interface_path(interface_identifier)
+    with open(file_path) as file_handler:
+        while True:
+            line = file_handler.readline()
+            if not line:
+                break
```

Comment:
nit, I think the `while True:` and these 3 lines could be replaced with `for line in file_handler:`